.github/workflows/lint-shared-workflows.yaml

```diff
@@ -1,16 +1,63 @@
 on:
   push:
-  pull_request:
 
 jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+      - name: Checkout
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
         with:
           ref: ${{ github.head_ref }}
 
-      - uses: creyD/prettier_action@31355f8eef017f8aeba2e0bc09d8502b13dbbad1 # v4.3.0
+      - name: Lint with Prettier
+        uses: creyD/prettier_action@31355f8eef017f8aeba2e0bc09d8502b13dbbad1 # v4.3.0
         with:
           dry: true
           prettier_options: "--check ."
+
+      # Lint .github/workflows/*
+      - name: Lint workflow files
+        uses: raven-actions/actionlint@789059c543ab20522fb3e7240794e13b0f69ad67 # v1.0.3
+
+  # A separate job so we can run in the `yq` container
+  lint-action-yaml:
+    runs-on: ubuntu-latest
+
+    container:
+      image: mikefarah/yq:4.43.1
+      # https://github.com/actions/checkout/issues/956
+      options: --user root
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+
+      - id: convert-action-yaml-to-json
+        name: Convert action YAMLS to JSON
+        run: |
+          set -ex
+
+          find . -name 'action.yaml' -o -name 'action.yml' | while read -r file; do
+            JSON_FILE="${file%.*}.json"
+
+            yq eval -o=j "$file" > "${JSON_FILE}"
+
+            # Save converted filenames to a file
+            echo "${JSON_FILE}" >> converted_files.txt
+          done
+
+          echo "Converted: $(tr '\n' ' ' < converted_files.txt)"
+
+          # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-of-a-multiline-string
+          {
+            echo 'converted_files<<EOF'
+            cat converted_files.txt
+            echo 'EOF'
+          } >> "${GITHUB_OUTPUT}"
+
+      - name: Validate action definitions
+        uses: ScratchAddons/validate-json-action@8f71e0683221310e32661c1b1634399858bde75f
+        with:
+          schema: https://json.schemastore.org/github-action.json
```

Comment:
Would this also work with a local JSONSchema file so that we can download and cache it?

Reply:
Yes, here you go: a66a17fd2b6ef0a5f170bdbcb1ff6afc9c67d183!

  1. [Without cache](https://github.com/grafana/shared-workflows/actions/runs/8988827934/job/24690531924)
  2. [With cache](https://github.com/grafana/shared-workflows/actions/runs/8988874065/job/24690679827)

Reply:
I realised with the previous one it wasn't saving the cache. Check the latest: now we explicitly restore it. More faff was required though: you can't directly overwrite a cache entry so I needed to add a step to first delete if we want to do that...